provider/datagen/src/lib.rs

```diff
@@ -125,13 +119,12 @@ pub fn keys<S: AsRef<str>>(strings: &[S]) -> Vec<ResourceKey> {
 /// # Ok(())
 /// # }
 /// ```
-pub fn keys_from_file<R: Read>(file: R) -> std::io::Result<Vec<ResourceKey>> {
-    let keys = BufReader::new(file)
-        .lines()
-        .collect::<std::io::Result<HashSet<String>>>()?;
+pub fn keys_from_bin<P: AsRef<Path>>(path: P) -> std::io::Result<Vec<ResourceKey>> {
+    let file = std::fs::read(path.as_ref())?;
```

Comment:
Nit: It would be more efficient to use BufReader and then call BufRead::[split](https://doc.rust-lang.org/stable/std/io/trait.BufRead.html#method.split)